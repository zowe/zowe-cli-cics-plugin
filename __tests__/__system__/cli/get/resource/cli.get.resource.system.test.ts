/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { runCliScript } from "../../../../__src__/TestUtils";
import { ITestEnvironment } from "../../../../__src__/environment/doc/response/ITestEnvironment";
import { TestEnvironment } from "../../../../__src__/environment/TestEnvironment";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment;
let regionName: string;
let host: string;
let port: number;
let user: string;
let password: string;

describe("cics get resource", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "get_resource_command",
            installPlugin: true,
            tempProfileTypes: ["cics"]
        });
        regionName = TEST_ENVIRONMENT.systemTestProperties.cmci.regionName;
        host = TEST_ENVIRONMENT.systemTestProperties.cmci.host;
        port = TEST_ENVIRONMENT.systemTestProperties.cmci.port;
        user = TEST_ENVIRONMENT.systemTestProperties.cmci.user;
        password = TEST_ENVIRONMENT.systemTestProperties.cmci.password;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/get_resource_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should be able to successfully get resources", async () => {
        const output = runCliScript(__dirname + "/__scripts__/get_resource.sh", TEST_ENVIRONMENT,
            ["CICSProgram", regionName]);
        const stderr = output.stderr.toString();
        const stdout = output.stdout.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(stdout).toContain("_keydata:");
    });

    it("should be able to successfully get resources using profile options", async () => {
        const output = runCliScript(__dirname + "/__scripts__/get_resource_fully_qualified.sh", TEST_ENVIRONMENT,
            ["CICSProgram",
                regionName,
                host,
                port,
                user,
                password]);
        const stderr = output.stderr.toString();
        const stdout = output.stdout.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(stdout).toContain("_keydata:");
    });
});
