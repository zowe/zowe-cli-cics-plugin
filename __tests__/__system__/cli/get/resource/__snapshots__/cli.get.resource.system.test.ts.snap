// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cics get resource should display the help 1`] = `
"===============cics PROFILE HELP===============

 COMMAND NAME
 ------------

   resource | res

 DESCRIPTION
 -----------

   Get resources (for example, programs or transactions) from CICS.

 USAGE
 -----

   zowe cics get resource <resourceName> [options]

 POSITIONAL ARGUMENTS
 --------------------

   resourceName		 (string)

      The name of the resource to get.

 OPTIONS
 -------

   --region-name  | --rn (string)

      The CICS region name from which to get the resources

   --cics-plex  | --cp (string)

      The name of the CICSPlex from which to get the resources

   --criteria  | -c (string)

      The criteria by which to filter the resource

   --parameter  | -p (string)

      The parameter by which to refine the resource

 CICS CONNECTION OPTIONS
 -----------------------

   --host  | -H (string)

      The CICS server host name.

   --port  | -P (number)

      The CICS server port.

      Default value: 443

   --user  | -u (string)

      Mainframe (CICS) user name, which can be the same as your TSO login.

   --password  | --pw (string)

      Mainframe (CICS) password, which can be the same as your TSO password.

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --protocol  | -o (string)

      Specifies CMCI protocol (http or https).

      Default value: http
      Allowed values: http, https

 PROFILE OPTIONS
 ---------------

   --cics-profile  | --cics-p (string)

      The name of a (cics) profile to load for this command execution.

 RESPONSE FORMAT OPTIONS
 -----------------------

   --response-format-filter  | --rff (array)

      Filter (include) fields in the response. Accepts an array of field/property
      names to include in the output response. You can filter JSON objects properties
      OR table columns/fields. In addition, you can use this option in conjunction
      with '--response-format-type' to reduce the output of a command to a single
      field/property or a list of a single field/property.

   --response-format-type  | --rft (string)

      The command response output format type. Must be one of the following:

      table: Formats output data as a table. Use this option when the output data is
      an array of homogeneous JSON objects. Each property of the object will become a
      column in the table.

      list: Formats output data as a list of strings. Can be used on any data type
      (JSON objects/arrays) are stringified and a new line is added after each entry
      in an array.

      object: Formats output data as a list of prettified objects (or single object).
      Can be used in place of \\"table\\" to change from tabular output to a list of
      prettified objects.

      string: Formats output data as a string. JSON objects/arrays are stringified.

      Allowed values: table, list, object, string

   --response-format-header  | --rfh (boolean)

      If \\"--response-format-type table\\" is specified, include the column headers in
      the output.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Get program resources from the region named MYREGION:

      $ zowe cics get resource CICSProgram --region-name MYREGION

   - Get local transaction resources from the region named
   MYREGION:

      $ zowe cics get resource CICSLocalTransaction --region-name MYREGION

   - Get local file resources from the region named MYREGION:

      $ zowe cics get resource CICSLocalFile --region-name MYREGION

   - Get program definition resources from the CSD group named
   GRP1 and the region named MYREGION:

      $ zowe cics get resource CICSDefinitionProgram --region-name MYREGION --parameter \\"CSDGROUP(GRP1)\\"

   - Get transaction definition resources from the CSD group
   named GRP1 and the region named MYREGION:

      $ zowe cics get resource CICSDefinitionTransaction --region-name MYREGION --parameter \\"CSDGROUP(GRP1)\\"

   - Get program resources that start with the name PRG from the
   region named MYREGION:

      $ zowe cics get resource CICSProgram --region-name MYREGION --criteria \\"PROGRAM=PRG*\\"

   - Get a local transaction resource named TRAN from the region
   named MYREGION:

      $ zowe cics get resource CICSLocalTransaction --region-name MYREGION --criteria \\"TRANID=TRAN\\"

   - Get program resources that start with the name MYPRG from
   the region named MYREGION and display various fields as a table:

      $ zowe cics get resource CICSProgram --region-name MYREGION --criteria \\"PROGRAM=MYPRG*\\" --rft table --rfh --rff program length status

"
`;
